src/core/random.hpp

```diff
@@ -83,6 +89,12 @@ void init_random_seed(int seed);
 
 inline double d_random() {
   using namespace Random;
+
+  if (!user_has_seeded && !unseeded_error_thrown) {
+    unseeded_error_thrown = true;
```

Comment:
there is a lot of code duplication here for every function that returns random numbers... also there is no need to throw if temperature is 0.0, e.g. the user only wants the damping part of a langevin thermostat